services/src/util/server.rs

```diff
@@ -4,18 +4,34 @@ use crate::util::config::get_config_element;
 
 use actix_http::body::{BoxBody, EitherBody, MessageBody};
 use actix_http::uri::PathAndQuery;
-use actix_http::HttpMessage;
+use actix_http::{Extensions, HttpMessage};
+
 use actix_web::dev::{ServiceFactory, ServiceRequest, ServiceResponse};
 use actix_web::error::{InternalError, JsonPayloadError, QueryPayloadError};
-use actix_web::{http, middleware, web, HttpResponse};
+use actix_web::{http, middleware, web, HttpRequest, HttpResponse};
+use futures::future::BoxFuture;
 use log::debug;
+
+#[cfg(target_os = "linux")]
+use actix_rt::net::TcpStream;
+#[cfg(target_os = "linux")]
+use nix::errno::Errno;
+#[cfg(target_os = "linux")]
+use nix::sys::socket::MsgFlags;
+use std::any::Any;
+#[cfg(target_os = "linux")]
+use std::num::NonZeroI32;
 use std::num::NonZeroUsize;
+#[cfg(target_os = "linux")]
+use std::os::unix::prelude::{AsRawFd, RawFd};
+use std::time::Duration;
+#[cfg(target_os = "linux")]
+use std::time::Instant;
```

Comment:
This is ugly to `cfg` that often.

You could…

 * put both into a module and one can just import "linux stuff".
 * import the stuff inside the `fn`

Reply:
8e9bbd185e7e2a422a5f9110eed35702276aef52